### PR TITLE
Fix loading operational mask when computing flux

### PR DESCRIPTION
**Pull request by dvida**
All raw collecting areas on operational systems have been computed without masks. This PR fixes that, hopefully resulting in correct fluxes.

### CODE REVIEW

**Comment by Cybis320** [comment]
Also, I think there is a similar issue with the platepar.

**Comment by dvida** [comment]
I added explicitly passing a platepar. However, a platepar is always available in the captured files directory, so that's fine. Otherwise, things would simply not work and the fluxes would not make any sense whatsoevery.

**Comment by Cybis320** [comment]
This still risks loading the wrong platepar, no?
```python
    if platepar is None:
        platepar = Platepar.Platepar()
        platepar.read(os.path.join(dir_path, config.platepar_name), use_flat=config.use_flat)
```
`platepar = Platepar.Platepar()` will create a default platepar with default values. If for whatever reasons there is no platepar in `dir_path`, those default values will not be updated. And there will be no platepar in dir_path is the recalibration fails.

**Comment by dvida** [comment]
I'm considering this a high priority fix. The platepar doesn't matter here as there will be no flux if the recalibrated platepars don't match the observations, the mask is the only critical thing here.

This works in my local tests. I'd like to merge this ASAP.

I'm recomputing all files on the server, hopefully we'll have updated flux profiles soon.

**Comment by markmac99** [comment]
I'll do my best to test this asap but see private message.

**Comment by dvida** [comment]
@Cybis320 if you look upstream in processNight, the command is not run if a platepar is not available or loaded.

However, I updated the code to make a check in the default dir second. I prefer the night dir first as the platepars might change over time so the current one might not always be appropriate for reprocessing.

**Comment by Cybis320** [comment]
> @Cybis320 if you look upstream in processNight, the command is not run if a platepar is not available or loaded.
> 
> However, I updated the code to make a check in the default dir second. I prefer the night dir first as the platepars might change over time so the current one might not always be appropriate for reprocessing.

Looks great! Thank you for humoring me :) Also, I can reuse the findBestPlatepar in my adsb code, so that's great too.

**Comment by Cybis320** [comment]
The findBestPlatepar is missing a couple 'return', maybe?

File: RMS/Formats/Platepar.py [hunk 1]
```diff
@@ -1470,6 +1470,38 @@ def __repr__(self):
...
+    platepar = Platepar.Platepar()
+    
+    if night_data_dir is not None:
+        night_platepar_path = os.path.join(night_data_dir, config.platepar_name)
+
+    default_platepar_path = os.path.join(config.config_file_path, config.platepar_name)
+
+    if (night_data_dir is not None) and os.path.exists(night_platepar_path):
+        platepar.read(night_platepar_path, use_flat=config.use_flat)
+
```
Comment (Cybis320) [inline RMS/Formats/Platepar.py:1497]: return platepar? and line 1500 too.

**Comment by dvida** [comment]
sorry, working through meetings and gaps in time. it should be fixed now. thanks!

**Comment by g7gpr** [comment]
Testing on Debian 10,11,12, Pi4, Pi5

File: Utils/RecomputeCollectionAreas.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        # Load the config file
+        config = loadConfigFromDirectory(['.'], dir_path)
+
+        # Load the platepar
+        platepar = Platepar.Platepar()
+        platepar.read(os.path.join(dir_path, config.platepar_name), use_flat=config.use_flat)
+
+        # Load the mask
+        mask = getMaskFile(dir_path, config)
+
```
Comment (Cybis320) [inline Utils/RecomputeCollectionAreas.py:26]: Are we not repeating the initial error here? Looking for platepar and mask in dir_path? Or is it meant to be run on a different folder structure?

Comment (dvida) [reply Utils/RecomputeCollectionAreas.py:26]: When running live, the code will already pass the correct mask and the platepar. When post-processing, RMS has to use the platepar and mask in the directory. E.g. imagine I'm reprocessing data from 2 years ago when the mask and platepar were different.

Comment (dvida) [reply Utils/RecomputeCollectionAreas.py:26]: I see the potential confusion - the directories this is made to reprocess are ArchivedFiles directories, not CapturedFiles.

Comment (Cybis320) [reply Utils/RecomputeCollectionAreas.py:26]: Oh, I see. Running this code on the Archived folder is fine. As long as the folder contains the appropriate platepar and mask without fail the code is fine.

**Comment by markmac99** [comment]
How should i be testing this? I checked out the PR to my test Pi4 and ran RMS.Reprocess on the ArchivedFiles folder for a dataset from May 2024, but the flux files were not updated.  The dataset contains 41 meteors.

**Comment by Cybis320** [comment]
I think mostly it's running reprocess on a CaptureFiles directory that needs testing. Whether it picks up the correct mask and platepar. For instance, if the correct mask is not all white but you see an all-white message in the console that would indicate that the wrong mask is used. 
Another interesting test is whether flux picks the correct platepar even if missing from the night dir. that's trickier to notice through. The way I noticed is that if the correct platepar is 1080p but flux picks a default 720p platepar, it generate a message. 
(Sorry typing on my cellphone)

**Comment by markmac99** [comment]
I've tested running RMS.Reprocess against both a CapturedFiles folder and an ArchivedFiles folder without error.  However, i am not sure what i should expect or how i'd know if it was broken! 

As there is no mask in the CapturedFiles folders, the mask in the RMS source folder must be being used.  I don't see any "all white" messages, even though the mask is the default all white. But this seems to be normal behaviour. 

There is a mask in the ArchivedFiles folder, but cannot confirm whether it was loaded in preference to the default mask as the code just prints "loaded mask; mask.bmp" at various places, without mentioning where it was loaded from :) But again this seems to be normal! 

The flux files were also unchanged (datestamp didn't update). I think i was expecting these to be updated, even if the mask was the same as before? 

So unfortunately i am not sure what this change is doing, or how i confirm it has worked.

**Comment by Cybis320** [comment]
I think, the `flux_col_areas_*.json` file should first be renamed for the file to be recalculated. The resulting json should be different if the original was computed with an incorrect all-white mask and the correct mask is non-white.

**Comment by dvida** [comment]
There were some issues with computing the masked ratio on some versions of numpy, I fixed that now.

How to test: 
Choose a camera that has a portion of the FOV masked out (preferably a good chunk of one corner). Run the normal capture. Next day, check the flux_col_areas*.json file and see if the computed area for the masked corner is zero.

The old version of the code ignored the mask, so the collecting areas were also full. That's probably why the flux had those deep dips.


I tested a reprocessing run locally without any issues. I'll test it on a live machine and hopefully merge it tomorrow.


The server is recomputing all previously computed areas.

**Pull request merged**